docs/qos_enforcement.md

```diff
@@ -0,0 +1,292 @@
+In this scenario, we will require for APN entities to only be specified through
+the orc8r.
+
+When the HSS is federated (and so `subscridberdb` is disabled), APN 
+configuration should be disabled.
+
+This adds the following requirement:
+
+6. Disabled configuration of APN entities for federated-HSS LTE networks
+
+## How We Will Change Magma
+
+### API Endpoints
+
+To enable configuration of QoS profiles and APNs, we will support the 
+following REST endpoints:
+
+```
+QoS Profiles
+
+GET     /lte/{network_id}/qos_profiles
+POST    /lte/{network_id}/qos_profiles
+
+GET     /lte/{network_id}/qos_profiles/{qos_name}
+PUT     /lte/{network_id}/qos_profiles/{qos_name}
+DELETE  /lte/{network_id}/qos_profiles/{qos_name}
+
+GET     /feg_lte/{network_id}/qos_profiles
+POST    /feg_lte/{network_id}/qos_profiles
+
+GET     /feg_lte/{network_id}/qos_profiles/{qos_name}
+PUT     /feg_lte/{network_id}/qos_profiles/{qos_name}
+DELETE  /feg_lte/{network_id}/qos_profiles/{qos_name}
+
+
+
+APN Profiles
+
+GET     /lte/{network_id}/apns
+POST    /lte/{network_id}/apns
+
+GET     /lte/{network_id}/apns/{apn_name}
+PUT     /lte/{network_id}/apns/{apn_name}
+DELETE  /lte/{network_id}/apns/{apn_name}
+
+
+
+APN Resources
+
+GET     /lte/{network_id}/gateways/{gateway_id}/resource_labels
+POST    /lte/{network_id}/gateways/{gateway_id}/resource_labels
+
+GET     /lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+PUT     /lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+DELETE  /lte/{network_id}/gateways/{gateway_id}/resource_labels/{resource_label}
+```
+
+
+
+### API Entity Definitions
+
+Entity definitions for QoS, APN, and subscribers will change. 
+The final state is summarized here:
+
+```
+qos_profile:
+  id: string
+  class_id: enum[0,255]
+  max_req_bw_dl: uint32
+  max_req_bw_ul: uint32
+  # Guaranteed bit rate
+  # Not enforced by Magma yet
+  gbr:
+    gbr_ul: uint32
+    gbr_dl: uint32
+  # Allocation and retention priority
+  # Not enforced by Magma yet
+  arp:
+    arp_priority_level: uint32
+    arp_preemption_capability: bool
+    arp_preemption_vulnerability: bool
+ # Aggregated Maximum bit rate
+ # for non-GBR bearers, this limits the aggregate bit rate
+ # Enforce this on pipelined
+ apn_ambr:
+   apn_ambr_dl: uint32
+   apn_ambr_ul: uint32
+```
+
+As an initial goal, we would like to support apn_ambr.
+
+```
+apn_configuration:
+  default_qos_profile: string
+  supported_qos_profiles: array[string]
+  # resource_label should match the name of a specified ApnResource
+  resource_label: string
+```
+
+Multiple QoS objects will be allowed to be associated to APN objects.
+
+One can be specified as the APN-level default policy. This is checked by 
+verifying a non-zero value for fields *apn_ambr_dl* and *apn_ambr_ul* for the 
+default QoS object.
+
+Additional QoS objects can be associated to the APN for additional resources 
+beyond the default policy, both physical and logical. 
```

Comment:
This may be confused with the apn_resource object defined below. Rephrasing to "Additional QoS objects can be associated with the APN to deliver dedicated QoS services for susbcribers' traffic".